### PR TITLE
Fixing a lot of string formatting issues

### PR DESCRIPTION
**Pull request by krousey**
- Improper format specifier (e.g. %s for bools or %s for ints)
- More or less parameters than format specifiers
- Not calling a formatting function when it should have (e.g. Error() instead of Errorf())

When there were too many parameters, I tried to add sensible format specifiers taking clues from the similar surrounding statements. When there were too few, I tried to find the intended thing to be put in the message.

If this is too big (like too many files), please let me know a sensible way to split it up.

### CODE REVIEW

File: pkg/kubelet/server_test.go [hunk 1]
```diff
@@ -448,7 +448,7 @@ func TestPodsInfo(t *testing.T) {
 	}
 	result := string(body)
 	if !strings.Contains(result, "pod level status currently unimplemented") {
-		t.Errorf("expected body contains %s, got %d", "pod level status currently unimplemented", result)
+		t.Errorf("expected body contains %s, got %s", "pod level status currently unimplemented", result)
```
Comment (vmarmol) [inline pkg/kubelet/server_test.go]: missing a second arg?

Comment (vmarmol) [inline pkg/kubelet/server_test.go]: Ah no nvm, can we just merge the strings :)

File: pkg/kubelet/server_test.go [hunk 2]
```diff
@@ -477,7 +477,7 @@ func TestHealthCheck(t *testing.T) {
 	}
 	result := string(body)
 	if !strings.Contains(result, "ok") {
-		t.Errorf("expected body contains %s, got %d", "ok", result)
+		t.Errorf("expected body contains %s, got %s", "ok", result)
```
Comment (vmarmol) [inline pkg/kubelet/server_test.go]: Same here.

File: pkg/kubelet/server_test.go [hunk 3]
```diff
@@ -987,7 +987,7 @@ func TestServePortForward(t *testing.T) {
 
 			p, err := strconv.ParseUint(test.port, 10, 16)
 			if err != nil {
-				t.Fatalf("%d: error parsing port string '%s': %v", i, port, err)
+				t.Fatalf("%d: error parsing port string '%s': %v", i, test.port, err)
```
Comment (vmarmol) [inline pkg/kubelet/server_test.go]: nit %q instead of '%s'

File: pkg/labels/selector.go [hunk 4]
```diff
@@ -445,7 +445,7 @@ func (p *Parser) parse() ([]Requirement, error) {
 		case IdentifierToken:
 			r, err := p.parseRequirement()
 			if err != nil {
-				return nil, fmt.Errorf("unable to parse requiremnt: ", err)
+				return nil, fmt.Errorf("unable to parse requiremnt: %s", err)
```
Comment (vmarmol) [inline pkg/labels/selector.go]: %v?

**Comment by vmarmol** [comment]
Thanks for the great cleanup @krousey! Out of curiosity, did you use a tool for this? Maybe we can run it more often :)

File: pkg/kubelet/server_test.go (hunk 3 above)
Comment (krousey) [inline pkg/kubelet/server_test.go]: Agreed, but there are least 70 more occurrences of this pattern in the code base. I don't mind fixing it, but I'd rather dedicate an entire PR to that one thing.

Comment (vmarmol) [inline pkg/kubelet/server_test.go]: SGTM

File: pkg/labels/selector.go (hunk 4 above)
Comment (krousey) [inline pkg/labels/selector.go]: Isn't it the same? Changed it anyway.

**Comment by krousey** [comment]
@vrmarmol go tool vet -all=0 -printf on each package.

File: pkg/labels/selector.go (hunk 4 above)
Comment (vmarmol) [inline pkg/labels/selector.go]: Yes, but we're standardizing on %v for errors.

**Comment by vmarmol** [comment]
Thanks for the pointer! This LGTM, lets squash the commits and merge.

**Comment by vmarmol** [comment]
LGTM and build is green. Thanks @krousey!

**Pull request merged**